test/extensions/tracers/xray/tracer_test.cc

```diff
@@ -183,6 +184,41 @@ TEST_F(XRayTracerTest, TraceIDFormatTest) {
   ASSERT_EQ(24, parts[2].length());
 }
 
+TEST(XRayDaemon, VerifyUdpPacketContents) {
+  NiceMock<Server::MockInstance> server;
+  Network::Test::UdpSyncPeer xray_fake_daemon(Network::Address::IpVersion::v4);
```

Comment:
Apologies but this needs to be parameterized on the supported IP types, as some people don't support v4 or run tests there. Please see the UDP example again for how to do this.

Reply:
No problem. I'm learning a lot this way.